iroh-net/src/lib.rs

```diff
@@ -103,8 +112,72 @@
 //! up the [`NodeId`] in the DNS system to find the addressing details.  This enables
 //! connecting using only the [`NodeId`] which is often more convenient and resilient.
 //!
+//! See [the discovery module] for more details.
+//!
+//!
+//! # Examples
+//!
+//! The central struct is the [`Endpoint`], which allows you to connect to other nodes:
+//!
+//! ```no_run
+//! use anyhow::Result;
+//! use iroh_net::{Endpoint, NodeAddr};
+//!
+//! async fn connect(addr: NodeAddr) -> Result<()> {
+//!     // The Endpoint is the central object that manages an iroh-net node.
+//!     let ep = Endpoint::builder().bind().await?;
+//!
+//!     // Establish a QUIC connection, open a bi-directional stream, exchange messages.
+//!     let conn = ep.connect(addr, b"hello-world").await?;
+//!     let (mut send_stream, mut recv_stream) = conn.open_bi().await?;
+//!     send_stream.write_all(b"hello").await?;
+//!     send_stream.finish()?;
+//!     let msg = recv_stream.read_to_end(10).await?;
+//!     println!("Server message: {}", String::from_utf8_lossy(&msg));
+//!
+//!     // Gracefully close the connection and endpoint.
+//!     conn.close(1u8.into(), b"done");
```

Comment:
I could delete this line as well.  I'm undecided.

Reply:
Don't you need this to make sure the conn.close has time to be sent before the entire program exits?

If anything I would have thought that the conn.close is the one that can be removed.

Reply:
Indeed, I meant to put that comment on the `conn.close()` line.

Reply:
No, you can leave out this `.close` call here.
I think Rüdiger was mixing this call up with the call to `ep.close` in the line below. That's the line that's *actually* waiting for the connection to close gracefully (because it calls `Endpoint::wait_idle` internally).

Reply:
I'm getting trolled by github.
This is when I'm in the "Conversation" tab:
![image](https://github.com/user-attachments/assets/155eaea1-c801-4846-9d4d-3e3b21687c38)

This is when I'm in the "Files" tab:
![image](https://github.com/user-attachments/assets/e8bed4e0-284a-4325-8ee7-2b861e72edd4)

Reply:
Anyways. I'm in favor of having a `conn.close` call here.
In the future I'd like the code to look like this instead:
```rs
conn.close(0u8.into(), b"done");
ep.all_closed().await?;
```
But that's irrelevant for *this* PR in particular.